docs/_includes/search-scripts.html

```diff
@@ -5,25 +5,29 @@
     window.searchStore = {
         {% for page in site.pages %}
         '{{ page.url | slugify }}': {
-            'title': '{{ page.title | xml_escape }}',
-            'description': {{ page.description | markdownify | strip_html | strip_newlines | jsonify }},
-            'usage': {{ page.usage | markdownify | strip_html | strip_newlines | jsonify }},
-            'accessibility': {{ page.usage | markdownify | strip_html | strip_newlines | jsonify }},
-            'url': '{{ page.url | xml_escape }}',
-
-            // Determine the maximum number of variations present overall.
-            {% if page.variations.size > variations_count %}
-                {% assign variations_count = page.variations | size  %}
-            {% endif %}
 
             // Cycle through the variation fields of this page.
-            {% for variation in page.variations %}
-                'variation_name{{ forloop.index }}': '{{ variation.variation_name | xml_escape }}',
-                'variation_description{{ forloop.index }}': {{ variation.variation_description | markdownify | strip_html | strip_newlines | jsonify }},
-                'variation_code_snippet{{ forloop.index }}': {{ variation.variation_code_snippet | markdownify | strip_html | strip_newlines | jsonify }}
-                {% unless forloop.last %},{% endunless %}
+            {% for variation_group in page.variation_groups %}
+                // Determine the maximum number of variations present overall.
+                {% if variation_group.variations.size > variations_count %}
+                    {% assign variations_count = variation_group.variations | size  %}
+                {% endif %}
+
+                {% for variation in variation_group.variations %}
+                    'variation_name{{ forloop.index }}': '{{ variation.variation_name | xml_escape }}',
+                    'variation_description{{ forloop.index }}': {{ variation.variation_description | markdownify | strip_html | strip_newlines | jsonify }},
+                    'variation_code_snippet{{ forloop.index }}': {{ variation.variation_code_snippet | markdownify | strip_html | strip_newlines | jsonify }},
```

Comment:
The `strip_html` filter results in many of the code snippet fields being empty on the search results page. For example, searching for `block__flush` doesn't return any results because the CSS classes get removed as part of `strip_html`.

Is there a safe way to include the HTML in the code snippets? Could we wrap `variation.variation_code_snippet` in backticks (<code>```</code>) and then pass it to `markdownify`, resulting in all the HTML/CSS surviving but being safely escaped?

Reply:
I `xml_escaped` it and put it in a template string.